agent_build_refactored/managed_packages/steps/install_build_dependencies.sh

```diff
@@ -0,0 +1,265 @@
+popd
+popd
+
+
+mkdir /tmp/build-help2man
+pushd /tmp/build-help2man
+curl -L "https://ftp.gnu.org/gnu/help2man/help2man-${HELP2MAN_VERSION}.tar.xz" > help2man.tar.xz
+tar -xvf help2man.tar.xz
+pushd "help2man-${HELP2MAN_VERSION}"
+./configure
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+mkdir /tmp/build-automake
+pushd /tmp/build-automake
+curl -L "https://ftp.gnu.org/gnu/automake/automake-${AUTOMAKE_VERSION}.tar.gz" > automake.tar.gz
+tar -xvf automake.tar.gz
+pushd "automake-${AUTOMAKE_VERSION}"
+./configure
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+mkdir /tmp/build-lzma
+pushd /tmp/build-lzma
+curl -L "https://tukaani.org/lzma/lzma-${LZMA_VERSION}.tar.gz" > lzma.tar.gz
+tar -xvf "lzma.tar.gz"
+pushd "lzma-${LZMA_VERSION}"
+./configure --prefix=/usr/local
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+mkdir /tmp/build-zlib
+pushd /tmp/build-zlib
+curl -L "https://www.zlib.net/zlib-${ZLIB_VERSION}.tar.gz" > zlib.tar.gz
+tar -xvf "zlib.tar.gz"
+pushd "zlib-${ZLIB_VERSION}"
+./configure --shared
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+mkdir /tmp/build-bzip
+pushd /tmp/build-bzip
+curl -L "https://sourceware.org/pub/bzip2/bzip2-${BZIP_VERSION}.tar.gz" > bzip2.tar.gz
+tar -xvf "bzip2.tar.gz"
+pushd "bzip2-${BZIP_VERSION}"
+make -f Makefile-libbz2_so -j "$(nproc)"
+make install
+cp -a libbz2.so* /usr/local/lib
+popd
+popd
+
+
+
+build_dir="/tmp/build"
+mkdir /tmp/build-util-linux
+pushd /tmp/build-util-linux
+curl -L "https://mirrors.edge.kernel.org/pub/linux/utils/util-linux/v2.38/util-linux-${UTIL_LINUX_VERSION}.tar.gz" > util-linux.tar.gz
+tar -xvf "util-linux.tar.gz"
+pushd "util-linux-${UTIL_LINUX_VERSION}"
+./configure --disable-all-programs --enable-libuuid --prefix=/usr/local
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+build_dir="/tmp/build"
+mkdir /tmp/build-ncurses
+pushd /tmp/build-ncurses
+curl -L "https://ftp.gnu.org/pub/gnu/ncurses/ncurses-${NCURSES_VERSION}.tar.gz" > ncurses.tar.gz
+tar -xvf "ncurses.tar.gz"
+pushd "ncurses-${NCURSES_VERSION}"
+./configure --with-shared --prefix=/usr/local
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+mkdir /tmp/build-libedit
+pushd /tmp/build-libedit
+curl -L "https://thrysoee.dk/editline/libedit-${LIBEDIT_VERSION}.tar.gz" > libedit.tar.gz
+tar -xvf "libedit.tar.gz"
+pushd "libedit-${LIBEDIT_VERSION}"
+./configure \
+  CFLAGS="-I/usr/local/include -I/usr/local/include/ncurses" \
+  LDFLAGS="-L/usr/local/lib -L/usr/local/lib64" \
+  LD_LIBRARY_PATH="/usr/local/lib:/usr/local/lib64:${LD_LIBRARY_PATH}"
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+mkdir /tmp/build-gdbm
+pushd /tmp/build-gdbm
+curl -L "https://ftp.gnu.org/gnu/gdbm/gdbm-${GDBM_VERSION}.tar.gz" > gdbm.tar.gz
+tar -xvf "gdbm.tar.gz"
+pushd "gdbm-${GDBM_VERSION}"
+./configure --enable-libgdbm-compat
+make -j "$(nproc)"
+make install
+popd
+popd
+
+
+mkdir /tmp/build-libffi
+pushd /tmp/build-libffi
+curl -L "https://codeload.github.com/libffi/libffi/tar.gz/v${LIBFFI_VERSION}" > libffi.tar.gz
```

Comment:
I know we use TLS transport, but long term we should also figure out how to make those downloads more secure - aka verifying checksums / PGP signatures (if available). That's basically what happens when using package manager (at least metadata signature is verified).

Reply:
Yes, that would be very nice to have. Will add it.